provider/lib/home/home_screen.dart

```diff
@@ -0,0 +1,137 @@
+
+  @override
+  void dispose() {
+    _tab.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(ProviderLocalizations.of(context).appTitle),
+        actions: <Widget>[
+          AnimatedBuilder(
+            animation: _tab,
+            builder: (_, __) => FilterButton(
+              isActive: _tab.value == _HomeScreenTab.todos,
+            ),
+          ),
+          const ExtraActionsButton(),
+        ],
+      ),
+      floatingActionButton: FloatingActionButton(
+        key: ArchSampleKeys.addTodoFab,
+        onPressed: () => Navigator.pushNamed(context, ArchSampleRoutes.addTodo),
+        tooltip: ArchSampleLocalizations.of(context).addTodo,
+        child: const Icon(Icons.add),
+      ),
+      body: Selector<TodoListModel, bool>(
+        selector: (context, model) => model.isLoading,
+        builder: (context, isLoading, _) {
+          if (isLoading) {
+            return Center(
+              child: CircularProgressIndicator(
+                key: ArchSampleKeys.todosLoading,
+              ),
+            );
+          }
+
+          return AnimatedBuilder(
+            animation: _tab,
+            builder: (context, _) {
+              switch (_tab.value) {
+                case _HomeScreenTab.stats:
+                  return const StatsView();
+                case _HomeScreenTab.todos:
+                default:
+                  return TodoListView(
+                    onRemove: (context, todo) {
+                      Provider.of<TodoListModel>(context, listen: false)
+                          .removeTodo(todo);
+                      _showUndoSnackbar(context, todo);
+                    },
+                  );
+              }
+            },
+          );
+        },
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        key: ArchSampleKeys.tabs,
+        currentIndex: _HomeScreenTab.values.indexOf(_tab.value),
+        onTap: (int index) {
+          _tab.value = _HomeScreenTab.values[index];
+        },
+        items: [
+          for (final tab in _HomeScreenTab.values)
+            BottomNavigationBarItem(
+              icon: Icon(tab.icon, key: tab.key),
+              title: Text(tab.title),
+            )
+        ],
+      ),
+    );
+  }
+
+  void _showUndoSnackbar(BuildContext context, Todo todo) {
+    Scaffold.of(context).showSnackBar(
+      SnackBar(
+        key: ArchSampleKeys.snackbar,
+        duration: const Duration(seconds: 2),
+        content: Text(
+          ArchSampleLocalizations.of(context).todoDeleted(todo.task),
+          maxLines: 1,
+          overflow: TextOverflow.ellipsis,
+        ),
+        action: SnackBarAction(
+          key: ArchSampleKeys.snackbarAction(todo.id),
+          label: ArchSampleLocalizations.of(context).undo,
+          onPressed: () =>
+              Provider.of<TodoListModel>(context, listen: false).addTodo(todo),
```

Comment:
Ah interesting, here the `addTodo` is done directly inside `Home` instead of being delegated to `ProviderApp` using a callback.

Reply:
Yah, might be good to be more consistent here and remove the callbacks or use them everywhere. I think i'll try removing them first, since that should be fine.

Reply:
Removed as many callbacks as I could, kept the ones that were necessary (the show snackbar ones)